src/silx/io/utils.py

```diff
@@ -598,6 +598,30 @@ def _open_local_file(filename):
     raise IOError("File '%s' can't be read as HDF5" % filename)
 
 
+def _open_url_with_h5pyd(url: str):
+    """Open the given url with h5pyd, raises an exception on failures
+
+    :param url: URL of a "file" served by HSDS
+    :returns: h5pyd.File instance
+    """
+    if h5pyd is None:
+        raise IOError(f"URL '{url}' unsupported. Try to install h5pyd.")
+
+    # Retrieve configured HSDS endpoint if any
+    default_endpoint = os.environ.get("H5SERV_ENDPOINT", None).rstrip("/")
+    if default_endpoint is None:
+        cfg = h5pyd.Config()
+        if "hs_endpoint" in cfg:
+            default_endpoint = cfg["hs_endpoint"].rstrip("/")
```

Comment:
If the environment variable is not set, rstrip operates on None, we can rstrip in the else branch

```suggestion
    default_endpoint = os.environ.get("H5SERV_ENDPOINT", None)
    if default_endpoint is None:
        cfg = h5pyd.Config()
        if "hs_endpoint" in cfg:
            default_endpoint = cfg["hs_endpoint"].rstrip("/")
    else:
        default_endpoint = default_endpoint.rstrip("/") 
```